File: fm-domain/src/main/java/io/zeitwert/fm/account/model/ObjAccountRepository.java

package io.zeitwert.fm.account.model;

import io.dddrive.core.obj.model.ObjRepository;
import io.zeitwert.fm.contact.model.ObjContactRepository;
import io.zeitwert.fm.dms.model.ObjDocumentRepository;

public interface ObjAccountRepository extends ObjRepository<ObjAccount> {

	ObjContactRepository getContactRepository();

	ObjDocumentRepository getDocumentRepository();

}

File: fm-domain/src/main/java/io/zeitwert/fm/account/model/ObjAccountRepository.kt
package io.zeitwert.fm.account.model

import dddrive.app.obj.model.ObjRepository
import io.zeitwert.fm.task.model.DocTaskRepository
import java.util.*

interface ObjAccountRepository : ObjRepository<ObjAccount> {

	fun getByKey(key: String): Optional<ObjAccount>

	val taskRepository: DocTaskRepository

}
